File: ai25btech11006/MATGEO/2.9.22/codes/main.py
import math

# Given magnitudes
norm_a = 1
norm_b = 2
norm_c = 3

# Since b and c are perpendicular, dot product is zero
dot_bc = 0

# Calculate v squared
v_squared = 9 * norm_a**2 + 4 * norm_b**2 + 4 * norm_c**2 - 8 * dot_bc

# Final magnitude
v_magnitude = math.sqrt(v_squared)

print(f"The magnitude |3a - 2b + 2c| is √{int(v_squared)} ≈ {v_magnitude:.4f}")

File: ai25btech11006/MATGEO/2.9.22/codes/main.c
#include <stdio.h>
#include <math.h>

int main() {
    // Given magnitudes
    double norm_a = 1.0;
    double norm_b = 2.0;
    double norm_c = 3.0;

    // Dot products not needed because they cancel or are zero
    double dot_bc = 0.0;

    // Calculate the magnitude squared of v = 3a - 2b + 2c
    double v_squared = 9 * norm_a * norm_a
                     + 4 * norm_b * norm_b
                     + 4 * norm_c * norm_c
                     - 12 * 0  // a.b cancels
                     + 12 * 0  // a.c cancels
                     - 8 * dot_bc;

    // Final magnitude
    double v_magnitude = sqrt(v_squared);

    printf("The magnitude |3a - 2b + 2c| is √%.0f ≈ %.4f\n", v_squared, v_magnitude);

    return 0;
}

